core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -766,37 +767,42 @@ def get_trained_params(self) -> dict:
         trained_params.update(self.params_trained)
         return trained_params
 
+    def get_params(self) -> dict:
+        """Get params of the model at the time of initialization"""
+        name = self.name
+        path = self.path_root
+        problem_type = self.problem_type
+        eval_metric = self.eval_metric
+        hyperparameters = self._user_params.copy()
+        if self._user_params_aux:
+            hyperparameters[AG_ARGS_FIT] = self._user_params_aux.copy()
+
+        args = dict(
+            path=path,
+            name=name,
+            problem_type=problem_type,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+        )
+
+        return args
+
     def convert_to_template(self):
         """After calling this function, returned model should be able to be fit as if it was new, as well as deep-copied."""
```

Comment:
Docstring might want to clarify model-name remains the same, so may overwrite the old model in the leaderboard?